gilbert_d2xy.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-2-Clause
+# Copyright (c) 2024 abetusk
+
+def gilbert_d2xy(idx,w,h):
+    """
+    Generalized Hilbert ('gilbert') space-filling curve for arbitrary-sized
+    2D rectangular grids. Takes a position along the gilbert curve and returns
+    its 2D (x,y) coordinate.
+    """
+
+    if w >= h:
+        return gilbert_d2xy_r(idx,0, 0,0, w,0, 0,h)
+    return gilbert_d2xy_r(idx,0, 0,0, 0,h, w,0)
+
+
+def sgn(x):
+    return -1 if x < 0 else (1 if x > 0 else 0)
+
+
+def inbounds(x,y, x_s,y_s, ax,ay, bx,by):
+
+    dx = ax+bx
+    dy = ay+by
+
+    if dx<0:
+        if (x>x_s) or (x<=(x_s+dx)): return False
+    else:
+        if (x<x_s) or (x>=(x_s+dx)): return False
+
+    if dy<0:
+        if (y>y_s) or (y<=(y_s+dy)): return False
+    else:
+        if (y<y_s) or (y>=(y_s+dy)): return False
+
+    return True
+
+
+def gilbert_d2xy_r( dst_idx, cur_idx, x,y, ax,ay, bx,by ):
+
+    w = abs(ax + ay)
+    h = abs(bx + by)
+
+    (dax, day) = (sgn(ax), sgn(ay)) # unit major direction
+    (dbx, dby) = (sgn(bx), sgn(by)) # unit orthogonal direction
+
+    dx = dax+dbx
```

Comment:
Please normalize whitespace, to match other code there should be spaces around `+`, `-`, `<=`, etc. (but not necessarily around `*`, `/`, this is supposed to mimic math typesetting).

Reply:
will do

Reply:
Note that this is violated in your own code:

https://github.com/jakubcerveny/gilbert/blob/4f9b20e8799a015e02a385b2d9f88dbc853fbec0/gilbert2d.py#L68-L69

The rule is any PEMDAS should have whitespace around it unless as a non-trivial function parameter?

For example, this is correct:

https://github.com/jakubcerveny/gilbert/blob/4f9b20e8799a015e02a385b2d9f88dbc853fbec0/gilbert2d.py#L25-L26

Because `abs` is a "trivial" function whereas `generate2d` is not?

Reply:
I didn't mean to be pedantic, sorry. I always put space after a comma, and usually around low priority operators, unless this is somehow counterproductive or there is limited space. I don't consider these to be hard rules, and I understand it's a matter of taste.